core/src/main/java/org/mule/api/registry/MuleTransportDescriptorService.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.api.registry;
+
+import org.mule.api.MuleContext;
+import org.mule.config.i18n.MessageFactory;
+import org.mule.util.SpiUtils;
+
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class MuleTransportDescriptorService implements TransportDescriptorService
+{
+
+    private final Map<String, TransportServiceDescriptorFactory> serviceDescriptorFactories = new ConcurrentHashMap<>();
+
+    @Override
+    public ServiceDescriptor getDescriptor(String transport, MuleContext muleContext, Properties overrides) throws ServiceException
+    {
+        TransportServiceDescriptorFactory transportServiceDescriptorFactory = serviceDescriptorFactories.get(transport);
+
+        if (transportServiceDescriptorFactory == null)
+        {
+            synchronized (serviceDescriptorFactories)
```

Comment:
Change serviceDescriptorFactories to be a Cache and create it using a CacheBuild with a CacheLoader implementation